components/SearchPage/index.tsx

```diff
@@ -0,0 +1,131 @@
+'use client';
+
+import type { Nullable, Results, Result } from '@orama/orama';
+import Link from 'next/link';
+import { useSearchParams } from 'next/navigation';
+import { useEffect, useState, type FC } from 'react';
+
+import type { SearchDoc } from '@/components/SearchBox/components/SearchBox';
+import { orama, highlighter } from '@/components/SearchBox/lib/orama';
+import { pathToBreadcrumbs } from '@/components/SearchBox/lib/utils';
+import { useBottomScrollListener } from '@/components/SearchPage/utils/useBottomScrollListener';
+
+import styles from './index.module.css';
+
+type SearchResults = Nullable<Results<SearchDoc>>;
+type Hit = Result<SearchDoc>;
+
+const SearchPage: FC = () => {
+  const searchParams = useSearchParams();
+  const [searchResults, setSearchResults] = useState<SearchResults>(null);
+  const [hits, setHits] = useState<Array<Hit>>([]);
+  const [offset, setOffset] = useState<number>(0);
+
+  const searchTerm = searchParams?.get('q');
+  const searchSection = searchParams?.get('section');
+
+  useBottomScrollListener(() => {
+    setOffset(offset => offset + 10);
+  });
+
+  useEffect(() => {
+    search(offset);
+  }, [offset]);
+
+  useEffect(() => {
+    setHits([]);
+    search(0);
+  }, [searchSection, searchTerm]);
+
+  const search = (resultsOffset = 0) => {
+    orama
+      .search({
+        term: searchTerm || '',
+        limit: 10,
+        offset: resultsOffset,
+        facets: {
+          siteSection: {},
+        },
+        ...filterBySection(),
+      })
+      .then(results => {
+        setSearchResults(results);
+        setHits(hits => [...hits, ...results.hits]);
+      })
+      .catch(console.log);
+  };
+
+  const facets = {
+    all: searchResults?.count ?? 0,
+    ...(searchResults?.facets?.siteSection?.values ?? {}),
+  };
+
+  const filterBySection = () => {
+    if (!searchSection || searchSection === 'all') {
+      return {};
+    }
+
+    return {
+      where: {
+        siteSection: {
+          eq: searchSection,
+        },
+      },
+    };
+  };
+
+  return (
+    <div className={styles.searchPageContainer}>
+      <div className={styles.searchTermContainer}>
+        <h1> {searchTerm} </h1>
+      </div>
+
+      <div className={styles.searchResultsColumns}>
+        <div className={styles.facetsColumn}>
+          {Object.keys(facets).map(facetName => (
+            <Link
+              key={facetName}
+              className={styles.searchResultsFacet}
+              href={`/en/search?q=${searchTerm}&section=${facetName}`}
+            >
+              {facetName}
+              <span className={styles.facetCount}>
+                ({facets[facetName as keyof typeof facets].toLocaleString('en')}
```

Comment:
This feels quite an odd typing coercion here

Reply:
TypeScript complains otherwise, but the looks clear (`facetName` value is one of the `facets` object keys)

Reply:
I'd probably assing this to a variable if possible.

Reply:
It's inside a `.map` so choices are:

1. Create a function to extract the facet number
2. Write a function body in the `.map` operation

I'd personally go with neither solutions as they cause unnecessary overhead